### PR TITLE
fix: augmented type generation

### PR DESCRIPTION
**Pull request by rflechtner**
## fixes KILTProtocol/ticket#2576

We're not the only ones with that problem, and from issues on the polkadot-js/api repo I took that type augmentation can no longer be run in cjs. We are now required to add the `--esm` flag when calling `ts-node` and to add the `"type":"module"` to the package.json. Fearing that this may interfere with the dual cjs/esm code generation in our repo, and realising we do not actually need code, just types from the augment-api package, I have refactored the build process only of this package to just produce types (and a few dummy files required for importing them from different environments.

Along the way, I have also tackled two FIXMEs that we've had sitting there for a while now. 

## How to test:

I tried building and publishing with `yalc` and using the resulting package in stakeboard, which works fine. Could be tried in other projects using this package as well.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [x] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Review by ntn-x2 — COMMENTED** [review]
Not too deep into all these JS/TS stuff, but it looks good. The only thing is that if I run `yarn build` in the root and then `yarn build:types` from within the augmentation package, the compiler still complains that `src/interfaces/extraDefs/types.ts` cannot find the name `PalletDidLookupLinkableAccountLinkableAccountId`, and in the diff I can see the old `FIXME` is gone. Is that expected? Is it just my own VSCode compiler issue?

File: packages/augment-api/src/interfaces/augment-api-rpc.ts [hunk 1]
```diff
@@ -275,7 +275,7 @@ declare module '@polkadot/rpc-core/types/jsonrpc' {
       /**
        * Returns the number of transactions sent from given address at given time (block number).
        **/
-      getTransactionCount: AugmentedRpc<(hash: H256 | string | Uint8Array, number?: BlockNumber | AnyNumber | Uint8Array) => Observable<U256>>;
+      getTransactionCount: AugmentedRpc<(address: H160 | string | Uint8Array, number?: BlockNumber | AnyNumber | Uint8Array) => Observable<U256>>;
```
[the remark this answers is not in the document]
Comment (rflechtner) [reply packages/augment-api/src/interfaces/augment-api-rpc.ts:278]: yeah we haven't been able to run type generation on this polkadot version, so this has been outdated until now

File: packages/augment-api/index.cjs
[the remark this answers is not in the document]
Comment (rflechtner) [reply packages/augment-api/index.cjs:1]: weirdly, I don't ?

[the remark this answers is not in the document]
Comment (rflechtner) [reply packages/augment-api/index.cjs:1]: added the header nevertheless

**Review by ntn-x2 — APPROVED** [review]

**Pull request merged**